enum.c

```diff
@@ -4154,6 +4155,7 @@ Init_Enumerable(void)
 #define rb_intern(str) rb_intern_const(str)
 
     rb_mEnumerable = rb_define_module("Enumerable");
+    rb_ensure_origin(rb_mEnumerable);
```

Comment:
Is this still needed?  This is a bit uneasy for me because it looks ad-hoc.  And as far as I understand, your patch now ensures the existence of origin whenever the module is `include`'ed, and the three modules (Enumerable, Comperable, and Kernel) are always `include`'ed, so I guess they are no longer needed.  In fact, the test passes without these calls to `rb_ensure_origin()`.

Reply:
No, these are no longer needed after the second patch.  I'll add another commit to remove them.